### PR TITLE
lightning(dm): enable to config concurrency when using lightning

### LINKED ISSUE #5505: User can set concurrency for lightning loader
### Is your feature request related to a problem?

currently lightning loader will use number of CPU cores as concurrency, but we have a pool-size configuration.

https://github.com/pingcap/tiflow/pull/5185#discussion_r873498039

### Describe the feature you'd like

as title

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by buchuitoudegou**
### What problem does this PR solve?

Issue Number: close #5505

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: dm/loader/lightning.go [hunk 1]
```diff
@@ -281,6 +281,10 @@ func (l *LightningLoader) restore(ctx context.Context) error {
 			return err
 		}
 		cfg := lcfg.NewConfig()
+		// TableConcurrency is adjusted to the value of RegionConcurrency
+		// when using TiDB backend.
+		// TODO: should we set the TableConcurrency separately.
+		cfg.App.RegionConcurrency = l.cfg.LoaderConfig.PoolSize
```
Comment (lance6716) [inline dm/loader/lightning.go:287]: seems put this after `cfg.LoadFromGlobal` is more similar to other logic

**Review by lance6716 — COMMENTED** [review]
not sure how we can add a test for it 🤔

File: dm/loader/lightning.go (hunk 1 above)
Comment (buchuitoudegou) [reply dm/loader/lightning.go:290]: [df5017c](https://github.com/pingcap/tiflow/pull/6328/commits/df5017c185bb829304899d16e8fabeebb2825c04)

File: dm/loader/lightning.go [hunk 2]
```diff
@@ -266,6 +266,44 @@ func (l *LightningLoader) runLightning(ctx context.Context, cfg *lcfg.Config) er
 	return err
 }
 
+func (l *LightningLoader) setLightningConfig() (*lcfg.Config, error) {
```
Comment (lance6716) [inline dm/loader/lightning.go:269]: it's getLightningConfig? (to caller)

**Review by lance6716 — APPROVED** [review]
please fix linter

File: dm/loader/lightning.go (hunk 2 above)
Comment (buchuitoudegou) [reply dm/loader/lightning.go]: 0463e8f

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**